docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,586 @@
+
+### Support for `bytes1`...`bytes32`
+
+CVL 2 supports the types `bytes1`, `bytes2`, ..., `bytes32`, as in Solidity.
+Number literals must be explicitly cast to these types using `to_bytesN`; for
+example:
+
+```cvl
+bytes32 x = to_bytes32(0);
+```
+
+There is no way to convert between these types and integer types (except for
+literals as just mentioned).
+
+```{todo}
+Update this if we add casting between `uint256` and `bytes32`.
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+(cvl2-bitwise)=
+### Changes for bitwise operations
+
+In CVL1, the exact details for bitwise operations (such as `&`, `|`, and `<<`) were not
+completely specified, especially for negative integers.
+
+In CVL 2, all bitwise operations (`&`, `|`, `~`, `>>`, `>>>`, `<<`, and `xor`)
+first convert to `uint256`, then perform the operations on the full 256-bit
+word, then convert back to the expected type.  Signed integer types use
+twos-complement encoding.
+
+The two right-shifts differ in how they treat signed integers.  `>>` is an
+arithmetic shift; it preserves the sign bit.  `>>>` is a logical shift; it pads
+the shifted word with zero.
+
+Bitwise operations cannot be performed on `mathint` values.
+
+### Conversion between `bytes<k>`, `address`, and integer types
+
+```{todo}
+finish
+```
+
+Removed features
+----------------
+
+As part of the transition to CVL 2.0, we have removed several language features
+that are no longer used.
+
+We have removed these features because we think they are no longer used and no
+longer useful.  If you find that you do need one of these features, contact
+Certora support.
+
+### Methods entries for sighashes
+
+In CVL 1, you could write a sighash instead of a method identifier in the
+`methods` block.  This feature is no longer supported.  You will need to have
+the name and argument types of the called method in order to provide an entry.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `invoke`, `sinvoke`, and `call`
+
+Older versions of CVL had special syntax for calling contract functions:
+ - `invoke f(args);` should be replaced with `f(args);`.
+ - `sinvoke f(args);` should be replaced with `f@withrevert(args);`
+ - `call f(args)` should be replaced with `f(args)`.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `static_assert` and `static_require`
+
+These deprecated aliases for `assert` and `require` are being removed; replace
+them with `assert` and `require`.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `invoke_fallback`
+
+The `invoke_fallback` syntax is no longer supported; there is no longer a way
+to directly invoke the fallback method.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+
+### Havocing `calldataarg` variables
+
+In CVL 1, you could write the following:
+
+```cvl
+calldataarg args; env e;
```

Comment:
Syntax highlighting does not recognize `calldataarg`

Reply:
This is a known problem that should be addressed elsewhere https://certora.atlassian.net/browse/DOC-214?atlOrigin=eyJpIjoiZWJiZWMyYTg5YmUyNDAzNTllMWE4NWI1ZWYxNDlkOWMiLCJwIjoiaiJ9